sbioapputils/app_runner/yaml_utils.py

```diff
@@ -0,0 +1,313 @@
+                                    'title': output_value['title']})
+        full_files.append(carousel_files)
+    return full_files
+    
+    
+def payload_from_config(yaml_dict):    
+    print("Generating payload from output config")
+    results_for_payload = {}
+    
+    if yaml_dict['output_settings'].get('images'):
+        results_for_payload['images'] = _generate_carousel(output_settings_dict=yaml_dict['output_settings'], result_type_key='images')
+        
+    if yaml_dict['output_settings'].get('figures'):
+        results_for_payload['figures'] = _generate_carousel(output_settings_dict=yaml_dict['output_settings'], result_type_key='figures')
+        
+    if yaml_dict['output_settings'].get('tables'):
+        results_for_payload['tables'] = _generate_carousel(output_settings_dict=yaml_dict['output_settings'], result_type_key='tables')
+        
+    if yaml_dict['output_settings'].get('download'):
+        full_files = []
+        for output_file in yaml_dict['output_settings']['download'].keys():
+            full_files.append({'file': yaml_dict['output_settings']['download'][output_file]['file'],
+                                 'title': yaml_dict['output_settings']['download'][output_file]['title']})
+        results_for_payload['download'] = full_files
+        
+    additional_artifacts = []
+    if yaml_dict['output_settings'].get('artifacts'):
+        for output_file, output_value in yaml_dict['output_settings']['artifacts'].items():
+            additional_artifacts.append(output_value['file'])        
+        
+    return results_for_payload, additional_artifacts
+
+
+def _generate_file_dict(file_list):
+    full_files = []
+    for file in file_list:
+        if file[0]=='/':
+            file = file[1:]
+        full_files.append({'file': file, 'title': file.split('/')[-1].split('.')[0]})
+    return full_files
+    
+
+def payload_from_folder(folder_loc, yaml_dict):
+    print("No payload config detected. Generating payload from output folder contents")
+    # Based on contents of a given folder instead
+    results_for_payload = {}
+    folder_contents = os.listdir(folder_loc)
+
+    tables = []
+    images = []
+    figures = []
+    additional_artifacts = []
+    for file in folder_contents:
+        file_ext = file.split('.')[-1]
+        if file_ext in ['csv', 'tsv', 'txt']:
+            tables.append(folder_loc + file)
+        elif file_ext in ['jpg', 'png']:
+            images.append(folder_loc + file)
+        elif file_ext in ['html']:
+            figures.append(folder_loc + file)
+        elif len(file.split('.')) > 1:
+            if os.path.isdir(file):
+                for sub_element in os.listdir(file):
+                    additional_artifacts.append(folder_loc + file + '/' + sub_element)
+            else:
+                additional_artifacts.append(folder_loc + file)
+    
+    #override get from folder if config provided in yaml
+    if yaml_dict['output_settings'].get('images'):
```

Comment:
yaml_dict['output_settings'] is used so many times, it should be a separate variable for readability

Reply:
Changed